common/main/java/com/couchbase/lite/internal/BaseImmutableReplicatorConfiguration.java

```diff
@@ -181,6 +182,22 @@ public void addConnectionOptions(@NonNull Map<String, Object> options) {
         final Map<String, Object> httpHeaders = new HashMap<>();
         // User-Agent:
         httpHeaders.put("User-Agent", CBLVersion.getUserAgent());
+
+        // Cookies:
+        final String cookieName = "Cookies";
+        if ((headers != null) && (!headers.isEmpty()) && headers.containsKey(cookieName)) {
+            //get current cookies value from option to add a new cookie
+            final String current = (String) options.get(C4Replicator.REPLICATOR_OPTION_COOKIES);
+            final StringBuffer cookieStr = (current == null) ? new StringBuffer() : new StringBuffer(current);
+
+            if (cookieStr.length() > 0) { cookieStr.append("; "); }
+            cookieStr.append(String.format(Locale.ENGLISH, "%s=%s", cookieName, headers.get(cookieName)));
```

Comment:
This seems to be going to a lot of work, in the case where the current cookies are null.  If that is the case, you don't need a string buffer and you don't need to append.  Just set the cookies!